verticapy/learn/vmodel.py

```diff
@@ -3186,20 +3192,96 @@ def predict(
                 ("cutoff", cutoff, [int, float]),
                 ("X", X, [list]),
                 ("vdf", vdf, [str, vDataFrame]),
+                ("inplace", inplace, [bool]),
             ],
         )
+        assert 0 <= cutoff <= 1, ParameterError(
+            "Incorrect parameter 'cutoff'.\nThe cutoff "
+            "must be between 0 and 1, inclusive."
+        )
         if isinstance(vdf, str):
             vdf = vdf_from_relation(relation=vdf)
         X = [quote_ident(elem) for elem in X]
-        name = (
-            "{}_".format(self.type) + "".join(ch for ch in self.name if ch.isalnum())
-            if not (name)
-            else name
+        if not (name):
+            name = gen_name([self.type, self.name])
+
+        # In Place
+        vdf_return = vdf if inplace else vdf.copy()
+
+        # Result
+        return vdf_return.eval(name, self.deploySQL(cutoff=cutoff, X=X))
+
+    # ---#
+    def predict_proba(
+        self,
+        vdf: Union[str, vDataFrame],
+        X: list = [],
+        name: str = "",
+        pos_label: Union[int, str, float] = None,
+        inplace: bool = True,
+    ):
+        """
+    ---------------------------------------------------------------------------
+    Returns the model's probabilities using the input relation.
+
+    Parameters
+    ----------
+    vdf: str/vDataFrame
+        Object to use to run the prediction. You can also specify a customized 
+        relation, but you must enclose it with an alias. For example, 
+        "(SELECT 1) x" is correct whereas, "(SELECT 1)" and "SELECT 1" are 
+        incorrect.
+    X: list, optional
+        List of the columns used to deploy the models. If empty, the model
+        predictors will be used.
+    name: str, optional
+        Name of the added vcolumn. If empty, a name will be generated.
+    pos_label: int/float/str, optional
+        Class label. For binary classification, this can be either 1 or 0.
+    inplace: bool, optional
+        If set to True, the prediction will be added to the vDataFrame.
+
+    Returns
+    -------
+    vDataFrame
+        the input object.
+        """
+        # Inititalization
+        if isinstance(X, str):
+            X = [X]
+        check_types(
+            [
+                ("name", name, [str]),
+                ("X", X, [list]),
+                ("vdf", vdf, [str, vDataFrame]),
+                ("inplace", inplace, [bool]),
```

Comment:
Don't you need check type for pos_label?

Reply:
Done;